io/js/src/main/scala/fs2/io/file/Path.scala

```diff
@@ -98,9 +98,21 @@ final case class Path private[file] (override val toString: String) extends Path
 object Path extends PathCompanionApi {
   private[file] val sep = facade.path.sep
 
+  /*
+   * NOTE: It seems like scala js does not rewrite this to a loop?
+   * Call stack limit is reached if there are too many separators.
+   */
+  @tailrec
+  private[file] def dropTrailingSep(path: String): String =
+    // Drop separator only if there is something else left
+    if (path.endsWith(sep) && path.length > sep.length)
+      dropTrailingSep(path.dropRight(sep.length))
+    else path
```

Comment:
It does actually 🤔  Here's what it's compiled to:

```js
$c_Lfs2_io_file_Path$.prototype.dropTrailingSep__T__T = (function(path) {
  while (true) {
    if ($f_T__endsWith__T__Z($n(path), this.Lfs2_io_file_Path$__f_sep)) {
      var this$1 = $n(path);
      var this$2 = $n(this.Lfs2_io_file_Path$__f_sep);
      var $$x2 = (this$1.length > this$2.length)
    } else {
      var $$x2 = false
    };
    if ($$x2) {
      var $$x1 = $m_sc_StringOps$();
      var x = path;
      var this$4 = $n(this.Lfs2_io_file_Path$__f_sep);
      path = $n($$x1).dropRight$extension__T__I__T(x, this$4.length)
    } else {
      break
    }
  };
  return path
});
```

Reply:
This passes for me. How were you getting the failure?
```scala
assertEquals(Path("/".repeat(Int.MaxValue/8)), Path("/"))
```

Reply:
Strange, I wrote this comment when I used `IndexedSeqView` instead of `String`.